### PR TITLE
Infoset Version 8

### PR DESCRIPTION
**Pull request by dknochen**
fixes 28msec/cellstore-nttdata#16
fixes 28msec/cellstore-nttdata#21
fixes 28msec/cellstore-nttdata#22
fixes 28msec/cellstore-nttdata#23

### CODE REVIEW

**Comment by dknochen** [comment]
Waiting for Version8 of infoset generator

**Comment by wcandillon** [comment]
Add examples: 
* Get element name of standard taxonomy by label
* Get element name of standard taxonomy by element name

File: swagger/queries.json [hunk 1]
```diff
@@ -2327,6 +2332,19 @@
                 "type": "string",
                 "paramType": "query"
             }, {
+                "name": "role",
```
Comment (wcandillon) [inline swagger/queries.json]: @gislenius Can you add an example of listReportElement() using the disclosure parameter?

File: queries/modules/io/28/modules/xbrl/concepts.jq [hunk 2]
```diff
@@ -61,21 +56,14 @@ declare variable $concepts:ANY_COMPONENT_LINK_ROLE as xs:string :=
...
-declare variable $concepts:VERBOSE_LABEL_ROLE as xs:string :=
-    "http://www.xbrl.org/2003/role/verboseLabel";
+declare variable $concepts:ALL_CONCEPT_NAMES as xs:string := "";
 
 (:~
- : A joker to ask for all concept names.
+ : A joker to ask for all concept labels.
  :)
-declare variable $concepts:ALL_CONCEPT_NAMES as xs:string := "";
+declare variable $concepts:ALL_CONCEPT_LABELS as xs:string := "";
```
Comment (dknochen) [inline queries/modules/io/28/modules/xbrl/concepts.jq]: could we not use an empty string for the ALL_* placeholders? Its hard to debug and in the end an empty label is more likely than e.g. ":all-labels:". Also, for example ":all-concepts:" isn't a valid concept name anyway. 
What do you think?

File: queries/public/api/facttable-for-component.jq [hunk 3]
```diff
@@ -88,7 +93,9 @@ let $components as object* :=
       $reportElement,
       $disclosure,
       $networkIdentifier,
-    $label)
+      $label[$profile-name ne "sec"],
+      $label[$profile-name eq "sec"]
```
Comment (dknochen) [inline queries/public/api/facttable-for-component.jq]: why not just once:
     $label[$profile-name]

File: queries/public/api/report-elements.jq [hunk 4]
```diff
@@ -145,14 +84,21 @@ let $entities as object* :=
...
-        then local:concepts-for-archives-and-labels(archives:aid($archives), $label[1])
-        else local:concepts-for-archives(archives:aid($archives), $name, $map, { OnlyNames: $onlyNames })
+    multiplexer:concepts(
+      $profile-name,
+      $archives,
+      $name,
+      $disclosure,
+      $role,
+      $label[$profile-name ne "sec"],
+      $label[$profile-name eq "sec"],
```
Comment (dknochen) [inline queries/public/api/report-elements.jq]: I understand this now ;). Would the full text index also work for profile "japan" and language=en? Or, is it only profile dependent?

File: queries/public/api/report-elements.jq [hunk 5]
```diff
@@ -182,22 +127,22 @@ let $result :=
...
-                          Language: $language,
-                          Value: $concept.Labels.$labelRole.$language
+                      Labels: backend:url("labels", {|
+                        {
+                          token: $token,
+                          concept: $original-name,
+                          aid: $archive,
+                          role: $role,
+                          format: $format,
+                          profile-name: $profile-name
```
Comment (dknochen) [inline queries/public/api/report-elements.jq]: I like this change. Though, should we add parameter labels as boolean to request labels inline the way it was?

File: queries/modules/io/28/modules/xbrl/labels.jq [hunk 6]
```diff
@@ -0,0 +1,232 @@
+jsoniq version "1.0";
```
Comment (dknochen) [inline queries/modules/io/28/modules/xbrl/labels.jq]: this cleanup makes so much sense :+1: :)

File: queries/modules/io/28/modules/xbrl/profiles/multiplexer.jq [hunk 7]
```diff
@@ -203,7 +205,8 @@ declare function multiplexer:components(
   $concept as string*,
   $disclosure as string*,
   $role as string*,
-  $label as string*) as object*
+  $exact-label as string*,
+  $full-text-label as string*) as object*
```
Comment (dknochen) [inline queries/modules/io/28/modules/xbrl/profiles/multiplexer.jq]: documentation needs to be updated

File: queries/modules/io/28/modules/xbrl/profiles/multiplexer.jq [hunk 8]
```diff
@@ -146,7 +149,8 @@ declare function multiplexer:components(
   $concept as string*,
   $disclosure as string*,
   $role as string*,
-  $label as string*,
+  $exact-label as string*,
+  $full-text-label as string*,
   $options as object?) as object*
```
Comment (dknochen) [inline queries/modules/io/28/modules/xbrl/profiles/multiplexer.jq]: documentation needs to be updated. LabelsOnly doesn't seem to exist any more.

File: swagger/queries.json [hunk 9]
```diff
@@ -2235,6 +2240,26 @@
                     "labelRole" : "http://www.xbrl.org/2003/role/label",
                     "format" : "xml"
                 }
+            }, {
+                "title": "Get label of standard taxonomy by element name",
```
Comment (dknochen) [inline swagger/queries.json]: example for https://github.com/28msec/cellstore-nttdata/issues/22

File: swagger/queries.json [hunk 10]
```diff
@@ -2235,6 +2240,26 @@
...
+                "description": "Get label of standard taxonomy by element name.",
+                "profile": ["japan"],
+                "parameters": {
+                    "aid": "STANDARD-TAXONOMY-2014",
+                    "format": "json",
+                    "profile-name": "japan",
+                    "reportElement": "jppfs-cor:CurrentAssets"
+                }
+            }, {
+                "title": "Get element name of standard taxonomy by label, as XML.",
```
Comment (dknochen) [inline swagger/queries.json]: note to myself: example for https://github.com/28msec/cellstore-nttdata/issues/21

File: queries/public/test/edinet/labels.jq [hunk 11]
```diff
@@ -21,5 +35,11 @@ test:check-all-success({
       function($b as item*) as item* { $b.Labels },
       test:get-expected-result("edinet/labels-expected4.jq"),
       { NoArrayOrder: true }
+    ),
+    taxonomy-fsa-label: test:invoke-and-assert-deep-equal(
+      "labels",
+      {aid:"STANDARD-TAXONOMY-2014", reportElement: ["fsa:FiscalYear", "fsa:FiscalPeriod", "fsa:FiscalPeriodType", "fsa:Submitted"] },
```
Comment (dknochen) [inline queries/public/test/edinet/labels.jq]: note to myself: test for https://github.com/28msec/cellstore-nttdata/issues/16

File: swagger/queries.json [hunk 12]
```diff
@@ -775,6 +775,12 @@
                 "type": "boolean",
                 "paramType": "query"
             }, {
+                "name": "language",
```
Comment (dknochen) [inline swagger/queries.json]: there seems to be no test nor example for this. Could you add either/or? Thanks.

https://github.com/28msec/cellstore-nttdata/issues/23

File: queries/modules/io/28/modules/xbrl/concepts.jq (hunk 2 above)
Comment (gislenius) [inline queries/modules/io/28/modules/xbrl/concepts.jq]: Of course we could, and that's what's done now :-)

File: queries/public/api/report-elements.jq (hunk 4 above)
Comment (gislenius) [inline queries/public/api/report-elements.jq]: For the moment, we don't support full text in Japan. We also need to migrate it for secxbrl as it's the old index.

File: queries/public/api/report-elements.jq (hunk 5 above)
Comment (gislenius) [inline queries/public/api/report-elements.jq]: It makes sense, but I think it would be nice to stick with flat content for now as much as we can and add this parameter later, what do you think? Flat content makes csv and xml serialization much easier.

File: queries/modules/io/28/modules/xbrl/labels.jq (hunk 6 above)
Comment (gislenius) [inline queries/modules/io/28/modules/xbrl/labels.jq]: Thanks ;-)

File: queries/modules/io/28/modules/xbrl/profiles/multiplexer.jq (hunk 8 above)
Comment (gislenius) [inline queries/modules/io/28/modules/xbrl/profiles/multiplexer.jq]: DONE

File: queries/modules/io/28/modules/xbrl/profiles/multiplexer.jq (hunk 7 above)
Comment (gislenius) [inline queries/modules/io/28/modules/xbrl/profiles/multiplexer.jq]: DONE

File: queries/modules/io/28/modules/xbrl/concepts.jq (hunk 2 above)
Comment (gislenius) [inline queries/modules/io/28/modules/xbrl/concepts.jq]: Ok, I get it now. I did it.

File: queries/public/api/report-elements.jq (hunk 5 above)
Comment (dknochen) [inline queries/public/api/report-elements.jq]: I agree. Nobody requested this, yet.

File: swagger/queries.json (hunk 1 above)
Comment (wcandillon) [inline swagger/queries.json]: Discard my comment above. The disclosure parameter is not part of this PR.

**Comment by dknochen** [comment]
@gislenius what about this comment? https://github.com/28msec/cellstore/pull/648/files#r24170188

**Comment by dknochen** [comment]
thanks for addressing the comments!

File: swagger/queries.json (hunk 12 above)
Comment (wcandillon) [inline swagger/queries.json]: example would be nice

**Comment by wcandillon** [comment]
@gislenius @dknochen If you can send me an example URL for
* Get element name of standard taxonomy by label
* Get element name of standard taxonomy by element name
I can take it from there. I can also add an example for the language parameter in `facttable-for-componement.jq`

**Comment by gislenius** [comment]
Hi William,

The two examples you requested were actually already here (in report-elements.jq). I added an example with the language parameter in facttable-for-component.jq.

As soon as the tests pass, anybody feel free to push the merge button and call it a release. ;-).

Kind regards,
Ghislain

**Comment by gislenius** [comment]
@dknochen the comment was removed by William.

**Comment by wcandillon** [comment]
Yes I will push the button. I'm putting final touches into the dashboard, if the isAbstract filter gets in, I think we can ship something really great.

**Comment by wcandillon** [comment]
@gislenius The following doesn't seem to return the appropriate result:
```bash
curl -X POST "http://edinet-apidoc2.28.io/v1/_queries/public/api/report-elements.jq?token=47ce6638-6193-4d21-afb6-532ef332bad1&format=json&aid=STANDARD-TAXONOMY-2014&name=jppfs-cor%3ACurrentAssets"
```
Can you confirm?

**Comment by gislenius** [comment]
Hi @wcandillon 
I don't see anything wrong with this link (I just had to replace the token with the dev token).
Kind regards,
Ghislain

**Comment by wcandillon** [comment]
@gislenius If I understand the API correctly, this link is asked to retrieve all report elements from the std taxonomy archive that is named jppfs-cor:ACurrentAssets. The response seems to return all of them no?
